shared-components/AlertBuilder.tsx

```diff
@@ -0,0 +1,124 @@
+import ReactDOM from "react-dom";
+import * as Stitches from "@stitches/react";
+
+// import { styled } from "@/shared-components/stitches.config";
+import { styled, css, Layout } from "@/shared-components/index";
+
+import Button from "./Button";
+import Span from "./Span";
+
+const StyledAlert = styled("div", {
+  display: "flex",
+  alignItems: "center",
+  justifyContent: "center",
+  width: "100vw",
+  height: "100vh",
+  position: "fixed",
+  left: 0,
+  top: 0,
+  zIndex: 999,
+  backgroundColor: "rgba(0,0,0,0.6)",
+  overflow: "hidden",
+});
+
+const AlertOverlay = styled(Layout.ColCenter, {
+  backgroundColor: "$white",
+  borderRadius: "8px",
+  margin: "0 16px",
+});
+
+interface AlertOptions {
+  message: React.ReactNode;
+  onClose?: () => void;
+  buttonMessage: React.ReactNode;
+  css?: Stitches.CSS;
+  buttonCSS?: Stitches.CSS;
+}
+
+class AlertBuilder {
+  private rootElement?: HTMLDivElement;
+  private lastTargetElement?: HTMLDivElement;
+
+  public show = (options: AlertOptions) => {
+    const { message, buttonMessage, onClose, css, buttonCSS } = options;
+    this.renderRootElement();
+    this.lastTargetElement = document.createElement("div");
+    ReactDOM.render(
+      <AlertMessage
+        message={message}
+        onRemove={this.removeRootMessage}
+        buttonMessage={buttonMessage}
+        css={css}
+        onClose={onClose}
+        buttonCSS={buttonCSS}
+      />,
+      this.lastTargetElement
+    );
+
+    this.rootElement?.appendChild(this.lastTargetElement);
+  };
+
+  private renderRootElement = () => {
+    const divElement = document.createElement("div");
+    divElement.className = `alert-container`;
+    document.body.style.overflow = "hidden";
+    document.body.appendChild(divElement);
+    this.rootElement = divElement;
+  };
+
+  private removeRootMessage = () => {
+    document.body.style.overflow = "unset";
+    this.rootElement?.remove();
+  };
+}
+
+export const Alert = new AlertBuilder();
+
+interface AlertMessageProps {
+  message: React.ReactNode;
+  onClose?: () => void;
+  onRemove: () => void;
+  css?: Stitches.CSS;
+  buttonMessage: React.ReactNode;
+  buttonCSS?: Stitches.CSS;
+}
+
+class AlertMessage extends React.Component<AlertMessageProps> {
+  render() {
+    const { onRemove, onClose, message, css, buttonMessage, buttonCSS } = this.props;
+    return (
+      <StyledAlert css={css}>
+        <AlertOverlay onClick={(e) => e.stopPropagation()}>
+          <Span
+            size={"h5"}
+            css={{
+              lineHeight: "24px",
+              marginTop: "24px",
+              textAlign: "center",
+              whiteSpace: "pre-line",
+            }}
+          >
+            {message}
+          </Span>
+          <Button
+            css={{
+              ...buttonCSS,
+              fontSize: "12px",
+              marginTop: "14px",
+              marginBottom: "24px",
+              padding: "7px 24px",
+              height: "32px",
+            }}
```

Comment:
```suggestion
            css={{
              fontSize: "12px",
              marginTop: "14px",
              marginBottom: "24px",
              padding: "7px 24px",
              height: "32px",
              ...buttonCSS,
            }}
```

위치를 아래로 내려주셔야지 설정해놓은 css 속성이 오버라이딩 됩니닷